src/test/ensemble_mapping_config_tests.cpp

```diff
@@ -197,3 +197,125 @@ TEST_F(PipelineWithInputOutputNameMappedModel, ReferingToOriginalOutputNameFails
 
     EXPECT_EQ(factory.createDefinition("pipeline", info, connections, managerWithDummyModel), StatusCode::INVALID_MISSING_OUTPUT);
 }
+
+TEST_F(PipelineWithInputOutputNameMappedModel, SuccessfullyReloadPipelineAfterAddingModelMapping) {
+    // Load models
+    auto modelConfig = DUMMY_MODEL_CONFIG;
+    modelConfig.setBasePath(modelPath);
+    ASSERT_EQ(managerWithDummyModel.reloadModelWithVersions(modelConfig), StatusCode::OK);
+
+    // Create pipeline definition
+    std::vector<NodeInfo> info{
+        {NodeKind::ENTRY, "request"},
+        {NodeKind::DL, "dummyA", "dummy"},
+        {NodeKind::DL, "dummyB", "dummy"},
+        {NodeKind::EXIT, "response"},
+    };
+    std::unordered_map<std::string, std::unordered_map<std::string, InputPairs>> connections;
+
+    connections["dummyA"] = {
+        {"request", {{"vector", "input_tensor"}}}};
+    connections["dummyB"] = {
+        {"dummyA", {{"output_tensor", "input_tensor"}}}};
+    connections["response"] = {
+        {"dummyB", {{"output_tensor", "response_tensor_name"}}}};
+
+    // Validation fails since mapping is expected
+    PipelineDefinition pd("UNUSED_NAME", info, connections);
+    auto status = pd.validate(managerWithDummyModel);
+    EXPECT_TRUE(status.getCode() == ovms::StatusCode::INVALID_MISSING_INPUT ||
+                status.getCode() == ovms::StatusCode::INVALID_MISSING_OUTPUT)
+        << status.string();
+    EXPECT_EQ(status.getCode(), ovms::StatusCode::INVALID_MISSING_OUTPUT) << status.string();
+
+    // Create mapping config for model
+    createConfigFileWithContent(R"({
+        "inputs": {"b": "input_tensor"},
+        "outputs": {"a": "output_tensor"}
+    })",
+        mappingConfigPath);
+
+    // reload pipeline definition after adding mapping
+    // we need to trigger reload - just adding model mapping won't do the trick
+    // hence setting nireq
+    modelConfig.setNireq(modelConfig.getNireq() + 1);
```

Comment:
Do we really need to do this hack? What about getting model instance and calling `reloadModel` on it with custom ModelConfig (setInputMapping/setOutputMapping)?

Reply:
This test could be integrating more functionalities later on (once we add PipelineFactory.reload(). If we call reloadModel on ModelInstance we may loose notification happening.